src/main/java/net/coderbot/iris/shaderpack/ShaderPack.java

```diff
@@ -86,21 +83,13 @@ private static ProgramSet loadOverrides(Path root, String subfolder, ShaderPrope
 		return Optional.of(properties);
 	}
 
-	public ProgramSet getProgramSet(DimensionId dimension) {
-		ProgramSet overrides;
-
-		switch (dimension) {
-			case OVERWORLD:
-				overrides = overworld;
-				break;
-			case NETHER:
-				overrides = nether;
-				break;
-			case END:
-				overrides = end;
-				break;
-			default:
-				throw new IllegalArgumentException("Unknown dimension " + dimension);
+	public ProgramSet getProgramSet(int dimension) {
+		ProgramSet overrides = null;
+
+		try {
+			overrides = loadOverrides(root, "world" + dimension, shaderProperties, this);
```

Comment:
We're trying to load all dimension overrides up front in order to avoid disk IO while compiling shaders. That should be possible using the dimensions ID map above.

This is also incorrect, because an unknown or unmatched dimension ID should not map to the world0 override it should map to the root directory.

Reply:
How would we store all this up front?